src/lib.rs

```diff
@@ -212,6 +214,45 @@ impl FromStr for Message {
     }
 }
 
+impl Serialize for Message {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: Serializer,
+    {
+        serializer.serialize_str(self.to_string().as_str())
+    }
+}
+
+struct MessageVisitor;
+
+impl<'de> Visitor<'de> for MessageVisitor {
+    type Value = Message;
+
+    fn expecting(&self, formatter: &mut fmt::Formatter) -> fmt::Result {
+        formatter.write_str("An EIP-4361 formatted message")
```

Comment:
Could you make this lowercase `an`? It makes the error message look nicer:
`"invalid type: map, expected An EIP-4361 message"`